### PR TITLE
working reach router in place! moved all components into their own files

### PR DESCRIPTION
**Pull request by aedward8**
Hi Avery Pls check this out! I was able to get basic react router up and running <3
Also refactored my code so components are in their own files

### CODE REVIEW

File: app/src/components/SignupForm.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      email: "",
+    },
+    validationSchema: Yup.object({
+      firstName: Yup.string()
+        .max(15, "Must be 15 characters or less")
+        .required("Required"),
+      lastName: Yup.string()
+        .max(20, "Must be 20 characters or less")
+        .required("Required"),
+      email: Yup.string().email("Invalid email address").required("Required"),
```
Comment (carbonsoda) [inline app/src/components/SignupForm.js:22]: This validation is really neat!!! 😎 Plus the form looks very sleek. It correctly doesn't allow emails like "avery@.com" or such.

File: app/src/components/AddTask.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
+
+import * as apiClient from "../apiClient";
+
+const AddTask = ({ loadTasks }) => {
+  const [task, setTask] = React.useState("");
+
+  const canAdd = task !== "";
```
Comment (carbonsoda) [inline app/src/components/AddTask.js:8]: Good that you use this check, so edge cases like "0" or "false" are still correctly let through.

File: package.json [hunk 3]
```diff
@@ -33,6 +33,7 @@
   },
   "homepage": "https://github.com/gsong/express-react-project-example#readme",
```
Comment (carbonsoda) [inline package.json:34]: Tiny tiny thing: don't forget to change the homepage to your repo 😄

File: app/src/App.js [hunk 4]
```diff
@@ -1,216 +1,22 @@
...
-          onBlur={formik.handleBlur}
-          value={formik.values.email}
-        />
-        {formik.touched.email && formik.errors.email ? (
-          <div>{formik.errors.email}</div>
-        ) : null}
-        <br />
-        <button type="submit">Submit</button>
-      </form>
+      <Router>
```
Comment (carbonsoda) [inline app/src/App.js:14]: If you plan on having the `<Nav />` on all pages, one idea is to stick it here instead! You'd put it right above `<Router>` ( [ex](https://github.com/carbonsoda/scribbler/blob/main/app/src/App.js#L22)).

File: app/src/components/Nav.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import * as React from "react";
+
+import { Link } from "@reach/router";
+
+const Nav = () => {
+  return (
+    <div>
```
Comment (carbonsoda) [inline app/src/components/Nav.js:7]: Small thing, maybe use `<nav>` over `<div>` here. It's more semantic and supported on all modern browsers (via [canIuse](https://caniuse.com/?search=nav)).

**Review by carbonsoda — COMMENTED** [review]
Nice work!! Especially on the separating out components part, looks very clean 😎 The only thing is that you installed reach-router in `package.json` but you need to include it in the `app/package.json` for Heroku to deploy it correctly I think.

File: app/src/components/AddTask.js (hunk 2 above)
Comment (aedward8) [reply app/src/components/AddTask.js:8]: i wish i could take credit but this is George's haha but i am learning ^_^

File: app/src/App.js (hunk 4 above)
Comment (aedward8) [reply app/src/App.js:14]: Ty for this suggestion!!

File: app/src/components/Nav.js (hunk 5 above)
Comment (aedward8) [reply app/src/components/Nav.js:7]: oh thank you so much!

**Comment by aedward8** [comment]
Changes have been implemented! Ty

**Pull request merged**